### PR TITLE
Query node input schema assets

### PR DESCRIPTION
**Pull request by ondratra**
Simplification of Asset representation + new redundant properties enabling queries important for Joystream.

### CODE REVIEW

File: query-node/schema.graphql [hunk 1]
```diff
@@ -65,55 +52,40 @@ type ChannelCategory @entity {
...
+"Asset availability representation"
+enum AssetAvailability {
+  ACCEPTED, # asset is available in storage
+  PENDING, # asset is being uploaded to storage
+  URL, # asset is referencing URL
+  INVALID, # invalid storage (meta)data used
+}
+
+"Asset representation"
+union Asset = AssetUrl | AssetStorage | AssetDeleted
```
Comment (bedeho) [inline query-node/schema.graphql]: I thought we were dropping `AssetDeleted`?

**Review by bedeho — CHANGES_REQUESTED** [review]

File: query-node/schema.graphql [hunk 2]
```diff
@@ -65,55 +52,40 @@ type ChannelCategory @entity {
...
 "Asset was deleted and is no longer available."
 type AssetDeleted @variant {
-  happenedIn: Block!
+  happenedIn: BigInt!
 }
 
-"Status of an asset upload"
-type AssetUploadStatus @variant {
+"Storage asset"
+type AssetStorage @variant {
```
Comment (kdembler) [inline query-node/schema.graphql]: Do we actually need this now? Couldn't we link directly to `AssetDataObject`? `AssetStorage` has an additional `happenedIn` but `AssetDataObject` has `addedAt` so I don't think we're losing any data?

File: query-node/schema.graphql [hunk 3]
```diff
@@ -300,12 +275,17 @@ type Video @entity {
...
+  happenedIn: BigInt!
 
   "Is video featured or not"
   isFeatured: Boolean!
 
   featured: FeaturedVideo @derivedFrom(field: "video")
+
+  # helper properties - redudant info enabling important queries for Joystream use-cases
+
+  is_thumbnail_available: AssetAvailability!
```
Comment (kdembler) [inline query-node/schema.graphql]: 1. Shouldn't we use `camelCase` here, like in other fields?
2. `is_..._available` suggests a boolean, available or not. Maybe `thumbnailAvailability` is better?

**Review by kdembler — CHANGES_REQUESTED** [review]
We're also missing the availability fields for channel assets

File: query-node/schema.graphql (hunk 1 above)
Comment (ondratra) [reply query-node/schema.graphql]: Yes, it was mistakenly kept there. Now the whole structure is simplified and flatten.

File: query-node/schema.graphql (hunk 2 above)
Comment (ondratra) [reply query-node/schema.graphql]: No, we don't.  Now the whole structure is simplified and flatten.

File: query-node/schema.graphql (hunk 3 above)
Comment (ondratra) [reply query-node/schema.graphql]: 1) Yes, that was a mistake.
2) Naming has been changed accordingly.

File: query-node/schema.graphql [hunk 4]
```diff
@@ -65,55 +52,34 @@ type ChannelCategory @entity {
...
-}
+  "Asset is being uploaded to storage"
+  PENDING,
 
-"Asset was deleted and is no longer available."
-type AssetDeleted @variant {
-  happenedIn: Block!
-}
+  "Asset is referencing URL"
+  URL,
```
Comment (kdembler) [inline query-node/schema.graphql]: I think we will be better off with removing `URL` as explicit state and use `ACCEPTED` for that? We will want to query entities with assets either accepted by storage or provided with URL and Hydra won't allow to express that (as there's no `OR` predicate) with this setup

File: query-node/schema.graphql [hunk 5]
```diff
@@ -57,35 +57,29 @@ type ChannelCategory @entity {
...
 
-"Storage asset"
-type AssetStorage @variant {
-  """
-  Data object in upload life-cycle.
-  If this is deleted, then set oldDataObject in its place if it is set and not rejected, otherwise union goes to Deleted.
-  """
-  dataObject: AssetDataObject!
+"Asset representation"
+type Asset @entity {
```
Comment (bedeho) [inline query-node/schema.graphql]: I don't think this is needed any longer.

**Review by bedeho — CHANGES_REQUESTED** [review]
Just one minor fix.

File: query-node/schema.graphql (hunk 4 above)
Comment (ondratra) [reply query-node/schema.graphql]: `URL` option has been removed.

File: query-node/schema.graphql (hunk 5 above)
Comment (ondratra) [reply query-node/schema.graphql]: `Asset` has been removed.

**Review by kdembler — APPROVED** [review]

**Review by bedeho — APPROVED** [review]

**Pull request merged**